ui/components/ui/delineator/delineator.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState } from 'react';
+import classnames from 'classnames';
+import {
+  BorderRadius,
+  AlignItems,
+  BackgroundColor,
+  IconColor,
+  Display,
+  JustifyContent,
+  FlexDirection,
+} from '../../../helpers/constants/design-system';
+import {
+  AvatarIcon,
+  Icon,
+  Box,
+  IconName,
+  IconSize,
+} from '../../component-library';
+import Preloader from '../icon/preloader/preloader-icon.component';
+import { DelineatorProps } from './delineator.types';
+import { getIconPropsByType, overrideTextComponentColorByType } from './utils';
+
+/**
+ * Renders the icon on the right based on the loading state.
+ *
+ * @param options - Options bag
+ * @param options.isLoading - Whether the delineator is in a loading state.
+ * @param options.isExpanded - Whether the delineator is expanded.
+ * @returns Either a loading spinner or an arrow icon.
+ */
+const ExpandableIcon = ({
+  isLoading,
+  isExpanded,
+}: {
+  isLoading: boolean;
+  isExpanded: boolean;
+}) => {
+  if (isLoading) {
+    return (
+      <div role="progressbar">
+        <Preloader size={16} />
+      </div>
+    );
+  }
+  return (
+    <Icon
+      name={isExpanded ? IconName.ArrowUp : IconName.ArrowDown}
+      size={IconSize.Sm}
+      color={IconColor.iconMuted}
+    />
+  );
+};
+
+export const Delineator: React.FC<DelineatorProps> = ({
+  children,
+  headerComponent,
+  iconName,
+  isCollapsable = true,
+  isExpanded: isExpandedProp,
+  isLoading = false,
+  onExpandChange,
+  type,
+  wrapperBoxProps,
+}) => {
```

Comment:
Will it be required to have props `isExpanded`, `onExpandChange ` one level above. Or they can be part of this component itself ?

Reply:
I think it will create unnecessary state management overload where this component is used if we leave the `expand` state to parent.